File: projects/octopus/src/main/java/octopus/api/plugin/PluginExecutor.java
package octopus.api.plugin;

import java.nio.file.Path;
import java.nio.file.Paths;

import org.json.JSONObject;

import octopus.OctopusEnvironment;
import octopus.server.pluginInterface.PluginLoader;

public class PluginExecutor {

	public Object executePlugin(String pluginName, String pluginClass)
	{
		return executePlugin(pluginName, pluginClass, null);
	}

	public Object executePlugin(String pluginName, String pluginClass, JSONObject settings)
	{
		Plugin plugin;

		if(!pluginName.equals("")){
			String pluginDir = OctopusEnvironment.PLUGINS_DIR.toString();
			Path pathToJar = Paths.get(pluginDir, pluginName).toAbsolutePath();
			plugin = PluginLoader.load(pathToJar, pluginClass);
		}else{
			plugin = PluginLoader.load(pluginClass);
		}

		if (plugin == null)
			throw new RuntimeException(
					"Error while loading plugin " + pluginName);

		try
		{
			if(settings != null)
				plugin.configure(settings);
			plugin.beforeExecution();
			plugin.execute();
			plugin.afterExecution();
			return plugin.result();
		} catch (Exception e)
		{
			System.out.println(e.getMessage());
			throw new RuntimeException(e.getMessage());
		}
	}

}

File: projects/octopus/python/octopus-mlutils/octopus/server/plugin_executor.py
import json

from octopus.server.server_command import ServerCommand

SERVER_HOST = 'localhost'
SERVER_PORT = '2480'

class PluginExecutor(object):
    def __init__(self, server_host = SERVER_HOST, server_port = SERVER_PORT):
        self.command = ServerCommand(server_host, server_port)

    def execute(self, pluginname, classname, settings=None):
        data = {"plugin": pluginname, "class": classname, "settings": settings}
        json_data = json.dumps(data)
        return self.post(json_data)

    def post(self, json_data):
        return self.command.execute_post_command("/executeplugin/", json_data)
